test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -1034,7 +1271,68 @@ public void Defect_MAGN_883()
             ViewModel.HomeSpace.Run();
             Assert.Pass("RunExpression should no longer crash (Defect_MAGN_883)");
         }
+        [Test]
+        public void WriteIntoExcelVerify()
+        {
+            // Write Into Excel sheet using WriteToFile
+            // Read Using ReadFromFile and verify the values are written correctly into Excel.
+
+            string testDir = TestDirectory;
+            string openPath = Path.Combine(testDir, @"core\excel\WriteFile.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            Assert.AreEqual(7, ViewModel.CurrentSpace.Nodes.Count);
+
+            var workspace = ViewModel.Model.CurrentWorkspace;
+            var filename = workspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", testDir);
+            Assert.AreEqual(7, ViewModel.CurrentSpace.Nodes.Count);
+            ViewModel.HomeSpace.Run();
+            //Write into Excel - {5,6,7,8,9,10}
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            
+            var data = new object[] { new object[] { 5 }, new object[] { 6 }, new object[] { 7 }, new object[] { 8 }, new object[] { 9 }, new object[] { 10 } };
+            AssertPreviewValue(watch.GUID.ToString(), data);
 
+            string openPath2 = Path.Combine(testDir, @"core\excel\ReadFile.dyn");
+            ViewModel.OpenCommand.Execute(openPath2);
+            Assert.AreEqual(5, ViewModel.CurrentSpace.Nodes.Count);
+
+            var filename2 = workspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename2.Value = filename2.Value.Replace(@"..\..\..\test", testDir);
+            ViewModel.HomeSpace.Run();
+            var watch2 = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("78370571-c7bc-4ed8-be2e-90c5be172767");
+            Assert.IsTrue(watch2.CachedValue.IsCollection);
+            //Read and verify the results are - {5,6,7,8,9,10}
+            var data2 = new object[] { new object[] { 5 }, new object[] { 6 }, new object[] { 7 }, new object[] { 8 }, new object[] { 9 }, new object[] { 10 } };
+            AssertPreviewValue(watch2.GUID.ToString(), data2);
+
+        }
+        [Test]
+        public void TestFormula()
+        {
+            string testDir = TestDirectory;
+            string openPath = Path.Combine(testDir, @"core\excel\formula.dyn");
```

Comment:
ive submitted the files here - https://github.com/monikaprabhu/Dynamo/commit/554878155adb111820d98d40b2f0e04325d6431e